src/Selda/Query/Class.purs

```diff
@@ -9,26 +9,26 @@ import Data.Either (Either, either)
 import Effect.Aff (Aff)
 import Effect.Aff.Class (class MonadAff, liftAff)
 import Foreign (Foreign)
-import Heterogeneous.Folding (class HFoldl, hfoldl)
 import Selda.Col (Col)
 import Selda.Table (Table)
 import Selda.Query.ShowStatement (class GenericShowInsert, genericShowInsert)
 import Selda.Query.Type (FullQuery)
-import Selda.Query.Utils (RecordToArrayForeign(..))
 import Type.Proxy (Proxy)
 
-class GenericQuery ∷ ∀ k. k → (Type → Type) → Row Type → Row Type → Constraint
-class Monad m <= GenericQuery b m i o | i → o, b → m where
+class GenericQuery ∷ ∀ k. k → Type → (Type -> Type) → (Type → Type) → Row Type → Row Type → Constraint
+class Monad m <= GenericQuery b decodeInput decodeOutput m i o | i → o, b → decodeInput decodeOutput m where
   genericQuery
     ∷ Proxy b
     → FullQuery b { | i }
+    → (decodeInput -> decodeOutput { | o })
```

Comment:
This is necessary because different backends have different ways of decoding things.

Reply:
these differences are not that serious - the common input could be just `Foreign` which is convertible to `Array Foreign` when needed. Similarly the error monad `decodeOutput` which could be just `m`